scripts/pre_commit_linter.py

```diff
@@ -505,6 +541,27 @@ def _check_newline_character(all_files):
     return summary_messages
 
 
+def _check_bad_pattern_in_file(filename, content, pattern):
+    """
+        Detects whether the given pattern is present in the file.
+        Args:
+            filename: str. Name of the file.
+            content: str. Contents of the file.
+            pattern: dict. Object containing details for the pattern
+                to be checked.
```

Comment:
Specify the keys and values (including types and descriptions of values).

Reply:
done.